mobile_rewards/src/server.rs

```diff
@@ -283,23 +293,39 @@ impl Server {
 
     async fn handle_rewards(
         &mut self,
-        reward_period: RewardPeriod,
-        last_reward_time: i64,
-        end_utc: i64,
+        reward_period: Range<u64>,
+        start_utc: DateTime<Utc>,
+        end_utc: DateTime<Utc>,
```

Comment:
this is going to introduce a bug when you persist this value back to the db by updating the Meta `last_reward_end_time`; you're calling `end_utc.to_string()` to update that value at the end of this function which is going to be something like `"2022-09-15 15:04:33.074973664 UTC"` so the next loop of the clock tick function it'll pass this string to your `get_time_range` function which is only accepting an i64 unix timestamp

Reply:
oooooooh nice catch, I'm going to re-work meta to fix issues like this